src/status_im2/contexts/chat/images_horizontal/view.cljs

```diff
@@ -0,0 +1,67 @@
+(ns status-im2.contexts.chat.images-horizontal.view
+  (:require [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]
+            [react-native.core :as rn]
+            [react-native.fast-image :as fast-image]
+            [utils.re-frame :as rf]
+            [react-native.safe-area :as safe-area]
+            [reagent.core :as reagent]
+            [oops.core :as oops]
+            [status-im2.contexts.chat.images-horizontal.style :as style]))
+
+(defn image
+  [message]
+  (let [shared-element-id (rf/sub [:shared-element-id])
+        window-width      (:width (rn/get-window))
+        height            (* (:image-height message) (/ window-width (:image-width message)))]
+    [fast-image/fast-image
+     {:source    {:uri (:image (:content message))}
+      :style     {:width         window-width
+                  :height        height
+                  :border-radius 12}
+      :native-ID (when (= shared-element-id (:message-id message)) :shared-element)}]))
+
+(defn get-item-layout
+  [_ index]
+  #js {:length (:width (rn/get-window)) :offset (* (:width (rn/get-window)) index) :index index})
```

Comment:
Suggestion for improved readability. It's common to break calculations over multiple lines to make it clear what's what.

```clojure
#js {:length (:width (rn/get-window))
     :index  index
     :offset (* (:width (rn/get-window))
                index)}
```

Reply:
i'm not sure #js works with dynamic values , it should be a static map as i remember, and i had a lot if issues with #js , better to use clj->js

Reply:
>Data in the form of a map {} or vector [] must follow the #js tag, which will be converted at compile-time to a JavaScript object or array, respectively.
>-- https://cljs.github.io/api/syntax/js-literal

Good point, it's a compile syntax sugar, but I don't see any problem here. I'm sure there can be problems with it, but do you have an example?

Reply:
so do you mean this code will work fine? unfortunately i don't have an example

Reply:
I know I'm sharing a contrived example below, but it shows that dynamic values are safe to be put inside the map. The code correctly returns true.

```clojure
(let [index   (rand-int 1000)
      length  (rand-int 100000000000)
      width   (rand-int 10000000)
      js-map  #js {:length length
                   :index  index
                   :offset (* width index)}
      clj-map {"length" length
               "index"  index
               "offset" (* width index)}]
  (= clj-map (js->clj js-map)))
```

Reply:
Both of `#js` and `clj->js` seem to give the same output: `#js {:length 390, :offset 390, :index 1}`